ethereum/executionclient/src/main/java/tech/pegasys/teku/ethereum/executionclient/schema/Response.java

```diff
@@ -13,51 +13,45 @@
 
 package tech.pegasys.teku.ethereum.executionclient.schema;
 
-import com.google.common.base.MoreObjects;
-import java.util.Objects;
 import java.util.Optional;
 import java.util.function.Function;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import tech.pegasys.teku.spec.SpecMilestone;
 
-public class Response<T> {
+public record Response<T>(
+    T payload, String errorMessage, boolean receivedAsSsz, boolean isUnsupportedMediaTypeError) {
   private static final Logger LOG = LogManager.getLogger();
 
-  private final T payload;
-  private final String errorMessage;
+  public static <T> Response<T> fromNullPayload() {
+    return new Response<>(null, null, false, false);
+  }
 
-  public Response(final T payload, final String errorMessage) {
-    this.payload = payload;
-    this.errorMessage = errorMessage;
+  public static <T> Response<T> fromPayloadReceivedAsSsz(final T payload) {
```

Comment:
maybe `fromSszPayload`

Reply:
but technically the payload has been already decoded at this stage. It indicates how you received the payload rather than the format of the payload passed here. I have mixed feelings